sqlsorcery.py

```diff
@@ -0,0 +1,100 @@
+from os import getenv
+from google.oauth2 import service_account
+import pandas as pd
+import pandas_gbq
+from sqlalchemy import create_engine, Table, MetaData
+from sqlalchemy.engine.url import URL
+from sqlalchemy.exc import NoSuchTableError
+from sqlalchemy.schema import DropTable
+
+
+class Connection:
+    def __init__(
+        self,
+        dialect=None,
+        user=None,
+        pwd=None,
+        server=None,
+        port=None,
+        db=None,
+        schema=None,
+        creds=None,
+    ):
+        self.dialect = dialect
+        self.schema = schema
+        self.db = db
+        self.options = {
+            "username": user,
+            "password": pwd,
+            "host": server,
+            "port": port,
+            "database": db,
+        }
+        self.creds = creds
+        self.helpers = {}
+        self.engine = self._get_engine()
+
+    def _cstr(self):
+        return URL(self.dialect, **self.options)
+
+    def _bigquery_options(self):
```

Comment:
Can we rename with a verb, like `_add_bigquery_options_if_needed`? It reads right now like it returns options (and always runs).